cmd/gardenlet/app/bootstrap.go

```diff
@@ -21,121 +21,65 @@ import (
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 	"github.com/gardener/gardener/pkg/gardenlet/apis/config"
 	"github.com/gardener/gardener/pkg/gardenlet/bootstrap"
-	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	bootstraputil "github.com/gardener/gardener/pkg/gardenlet/bootstrap/util"
 
 	"github.com/sirupsen/logrus"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	certificatesv1beta1client "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
-	"k8s.io/client-go/tools/clientcmd"
-	"k8s.io/client-go/util/keyutil"
 	componentbaseconfig "k8s.io/component-base/config"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// bootstrapKubeconfig retrieves an already existing kubeconfig for the Garden cluster from the Seed or bootstraps a new one
 func bootstrapKubeconfig(
 	ctx context.Context,
 	logger *logrus.Logger,
-	gardenClientConnection *config.GardenClientConnection,
+	gardenClientConnection config.GardenClientConnection,
 	seedClientConnection componentbaseconfig.ClientConnectionConfiguration,
 	seedConfig *config.SeedConfig,
 ) (
 	[]byte,
-	string,
-	string,
+	*string,
+	*string,
 	error,
 ) {
 	seedRESTCfg, err := kubernetes.RESTConfigFromClientConnectionConfiguration(&seedClientConnection, nil)
 	if err != nil {
-		return nil, "", "", err
+		return nil, nil, nil, err
 	}
-	k8sSeedClient, err := kubernetes.NewWithConfig(kubernetes.WithRESTConfig(seedRESTCfg))
+
+	seedClient, err := kubernetes.NewWithConfig(kubernetes.WithRESTConfig(seedRESTCfg))
```

Comment:
The `seedClient`'s cached is not started here, so any calls to `seedClient.Client().Get()` will return an error (if the `CachedRuntimeClients` feature gate is enabled.
You should either use the `DirectClient()` here or (what I actually prefer) construct a plain controller-runtime client with `client.New()` as you anyways don't need the other stuff contained in our client set.
```suggestion
	seedClient, err := client.New(seedRESTCfg, client.Options{})
```